.ci/Jenkinsfile

```
@@ -0,0 +1,110 @@
+// specific language governing permissions and limitations
+// under the License.
+
+@Library('apm@current') _
+
+pipeline {
+   agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    HOME = "${env.WORKSPACE}"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    JOB_GIT_CREDENTIALS = "f6c7695a-671e-4f4f-a331-acdce44ff9ba"
+    PATH = "${env.PATH}:${env.WORKSPACE}/bin"
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  parameters {
+    booleanParam(name: 'Do_Send_PR', defaultValue: false, description: 'Allows to execute this pipeline in dry run mode, without sending a PR.')
+  }
+  stages {
+    stage('Initializing'){
+      options { skipDefaultCheckout() }
+      stages {
+        stage('Checkout'){
+          options { skipDefaultCheckout() }
+          steps {
+            deleteDir()
+            gitCheckout(basedir: "${BASE_DIR}",
+              branch: "master",
+              repo: "git@github.com:elastic/${REPO}.git",
+              credentialsId: "${JOB_GIT_CREDENTIALS}"
+            )
+            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+          }
+        }
+        stage('Send Pull Request'){
+          options {
+            skipDefaultCheckout()
+            warnError('Pull Requests to APM agents failed')
+          }
+          environment {
+            DO_SEND_PR = "${params.Do_Send_PR}"
+          }
+          steps {
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              script {
+                def agents = readYaml(file: '.ci/.jenkins-agents.yml')
+                def parallelTasks = [:]
+                agents['agents'].each { agent ->
+                  parallelTasks["apm-agent-${agent.NAME}"] = generateStepForAgent(agent: "${agent.NAME}", featuresPath: "${agent.FEATURES_PATH}")
+                }
+
+                parallel(parallelTasks)
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def generateStepForAgent(Map params = [:]){
+  def agent = params.get('agent')
+  def featuresPath = params.get('featuresPath')
+  log(level: 'INFO', text: "agent=${agent} featuresPath=${featuresPath}")
+  return {
+    node('linux && immutable') {
+      try {
+        deleteDir()
+        unstash 'source'
+        dir("${BASE_DIR}"){
+          sh script: '.ci/scripts/install-dependencies.sh', label: "Install dependencies"
+          sh script: """.ci/scripts/send-pr.sh "${agent}" "${featuresPath}" """, label: "Send Pull Request for apm-agent-${agent}"
+        }
+      } catch(e) {
+        error(e.toString())
+      }
```

I do not understand why you catch an error to launch the error, try/catch is the last result, it is better to use catchError step, it allows you to control the result of the stage and the build, in the following example it would mark the stage as Unstable and the build as success

```suggestion
      catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
          deleteDir()
          unstash 'source'
            dir("${BASE_DIR}"){
              sh script: '.ci/scripts/install-dependencies.sh', label: "Install dependencies"
              sh script: """.ci/scripts/send-pr.sh "${agent}" "${featuresPath}" """, label: "Send Pull Request for apm-agent-${agent}"
            }
          }
```

---

It's a copy paste 🤔 I will update it here AND in the original source